<?xml version="1.0"?>
<launch>
    <arg name="rvizconfig" default="$(find cam_lecture)/rviz/orb.rviz" />

    <include file="$(find realsense2_camera)/launch/rs_rgbd.launch"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="link_broadcaster" args="0.05 0 0.1 0 0 0 1 base_link camera_link" />

    <group ns="/camera/color">
        <node name="camera_info_modifier" pkg="hard_lecture" type="camera_info_modifier" output="screen">
            <!-- distance color and depth camera on realsense d435 is 0.015[m] -->
            <param name="baseline_x" value="-0.015"/>
        </node>
    </group>


    <node name="orb_slam2_rgbd" pkg="orb_slam2_ros" type="orb_slam2_ros_rgbd" output="screen">

        <remap from="/camera/rgb/image_raw" to="/camera/color/image_raw" />
        <remap from="/camera/rgb/camera_info" to="/camera/color/camera_info_modified" />
        <remap from="/camera/depth_registered/image_raw" to="/camera/aligned_depth_to_color/image_raw" />

        <param name="publish_pointcloud" type="bool" value="true" />
        <param name="publish_pose" type="bool" value="true" />
        <param name="localize_only" type="bool" value="false" />
        <param name="reset_map" type="bool" value="false" />

        <!-- static parameters -->
        <param name="load_map" type="bool" value="false" />
        <param name="map_file" type="string" value="map.bin" />
        <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

        <param name="pointcloud_frame_id" type="string" value="map" />
        <param name="camera_frame_id" type="string" value="camera_link" />
        <param name="min_num_kf_in_map" type="int" value="5" />

        <!-- ORB parameters -->
        <param name="/ORBextractor/nFeatures" type="int" value="1000" />
        <param name="/ORBextractor/scaleFactor" type="double" value="1.2" />
        <param name="/ORBextractor/nLevels" type="int" value="8" />
        <param name="/ORBextractor/iniThFAST" type="int" value="20" />
        <param name="/ORBextractor/minThFAST" type="int" value="7" />

        <!-- Camera parameters -->
        <!-- Camera frames per second -->
        <param name="camera_fps" type="int" value="30" />
        <!-- Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale) -->
        <param name="camera_rgb_encoding" type="bool" value="true" />
        <!-- Close/Far threshold. Baseline times. -->
        <param name="ThDepth" type="double" value="40.0" />
        <!-- Deptmap values factor (what pixel value in the depth image corresponds to 1m) -->
        <param name="depth_map_factor" type="double" value="1000.0" />

        <!-- Camera calibration parameters -->
        <!--If the node should wait for a camera_info topic to take the camera calibration data-->
        <param name="load_calibration_from_cam" type="bool" value="true" />
  </node>

    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" /> -->
</launch>
